mindmeld/components/entity_resolver.py

```diff
@@ -42,60 +49,295 @@
     load_index,
     resolve_es_config_for_version,
 )
+from ._util import _is_module_available, _get_module_or_attr
+from .. import path
+from ..core import Entity
+from ..exceptions import EntityResolverConnectionError, EntityResolverError
 
 logger = logging.getLogger(__name__)
 
+DEFAULT_ENTITY_RESOLVER_MODEL_CONFIGS = {
+    "exact_match": {
+        'model_type': 'exact_match',
+        "model_settings": {
+            "augment_lower_case": False
+        }
+    },
+    "text_relevance": {
+        'model_type': 'text_relevance',
+        # 'phonetic_match_types': ["double_metaphone"],
+    },
+    "sbert_cosine_similarity": {
+        'model_type': 'sbert_cosine_similarity',
+        "model_settings": {
+            "pretrained_name_or_abspath": "distilbert-base-nli-stsb-mean-tokens",
+            "batch_size": 16,
+            "concat_last_n_layers": 4,
+            "normalize_token_embs": True,
+            "bert_output_type": "mean",
+            "augment_lower_case": False,
+            "quantize_model": True,
+        }
+    },
+    "tfidf_cosine_similarity": {
+        'model_type': 'tfidf_cosine_similarity',
+        "model_settings": {
+            "augment_lower_case": True,
+            "augment_normalized": False,
+            "augment_title_case": False
+        }
+    },
+}
 
-class EntityResolver:
-    """An entity resolver is used to resolve entities in a given query to their canonical values
-    (usually linked to specific entries in a knowledge base).
-    """
 
-    # prefix for Elasticsearch indices used to store synonyms for entity resolution
-    ES_SYNONYM_INDEX_PREFIX = "synonym"
-    """The prefix of the ES index."""
+class EntityResolver:
 
-    def __init__(
-        self, app_path, resource_loader, entity_type, es_host=None, es_client=None
-    ):
-        """Initializes an entity resolver
+    @staticmethod
+    def validate_resolver_name(name):
+        if name not in ENTITY_RESOLVER_MODEL_TYPES:
+            msg = "Expected 'model_type' in ENTITY_RESOLVER_CONFIG among {!r}"
+            raise Exception(msg.format(ENTITY_RESOLVER_MODEL_TYPES))
+        if name == "sbert_cosine_similarity" and not _is_module_available("sentence_transformers"):
+            raise ImportError(
+                "Must install the extra [bert] to use the built in embbedder for entity "
```

Comment:
maybe enumerate the whole command `pip install mindmeld[bert]` in the instruction here.